DisruptorUnity3d/Assets/RingBuffer.cs

```diff
@@ -47,13 +47,13 @@ public int Capacity
         /// <returns>The next available item</returns>
         public T Dequeue()
         {
-            var next = _consumerCursor.ReadFullFence() + 1;
-            while (_producerCursor.ReadFullFence() < next)
+            var next = _consumerCursor.ReadAcquireFence() + 1;
+            while (_producerCursor.ReadAcquireFence() < next) // makes sure we read the data from _entries after we have read the producer cursor
             {
                 Thread.SpinWait(1);
             }
             var result = this[next];
-            _consumerCursor.WriteFullFence(next);
+            _consumerCursor.WriteReleaseFence(next); // makes sure we read the data from _entries before we update the consumer cursor
```

Comment:
I noticed that. However you don't use the correct pattern. You mast write-release the cursor in order to make sure that the write to the array has completed. You write unfenced.